devicetypes/smartthings/vision-in-wall-2relays-switch.src/vision-in-wall-2relays-switch.groovy

```diff
@@ -0,0 +1,294 @@
+/**
+ *  Vision In-Wall 2Relays Switch (Models: ZL7435xx-5)
+ *
+ *  Author:
+ *    Lan, Kuo Wei 
+ *
+ *  Product Link: 
+ *    http://www.visionsecurity.com.tw/index.php?option=product&lang=en&task=pageinfo&id=335&belongid=334&index=0
+*/
+metadata {
+	definition(
+		name: "Vision In-Wall 2Relays Switch", 
+		namespace: "SmartThings", 
+		author: "Lan, Kuo Wei", 
+		ocfDeviceType: "oic.d.switch", 
+		mnmn: "SmartThings", 
+		vid: "generic-switch"
+	) {
+		capability "Actuator"
+		capability "Health Check"
+		capability "Refresh"
+		capability "Switch"
+		capability "Configuration"
+		// This DTH uses 2 switch endpoints. Parent DTH controls endpoint 1 so please use '1' at the end of deviceJoinName
+		// Child device (isComponent : false) representing endpoint 2 will substitute 1 with 2 for easier identification.
+		fingerprint manufacturer: "0109", prod: "2017", model: "171B", deviceJoinName: "Vision 2Relays Switch 1"
+		fingerprint manufacturer: "0109", prod: "2017", model: "171C", deviceJoinName: "Vision 2Relays Switch 1"
+	}
+
+	preferences {
+		parameterMap.each {
+			input (title: it.name, description: it.description, type: "paragraph", element: "paragraph")
+			switch(it.type) {
+				case "boolean":
+					input(type: "paragraph", element: "paragraph", description: "Option enabled: ${it.activeDescription}\n" +
+						"Option disabled: ${it.inactiveDescription}"
+					)
+					input(name: it.key, type: "boolean", title: "Enable", defaultValue: it.defaultValue == it.activeOption, required: false)
+					break
+					
+				case "enum":
+					input(name: it.key, title: "Select", type: "enum", options: it.values, defaultValue: it.defaultValue, required: false)
+					break
+			}	
+		}
+	}
+}
+
+def installed() {
+	def componentLabel
+	if (device.displayName.endsWith('1')) {
+		componentLabel = "${device.displayName[0..-2]}2"
+	} else {		
+		componentLabel = "$device.displayName 2"
+	}
+	try {
+		String dni = "${device.deviceNetworkId}-ep2"
+		addChildDevice("Z-Wave Binary Switch Endpoint", dni, device.hub.id,
+			[completedSetup: true, label: "${componentLabel}", isComponent: false])
+		log.debug "Endpoint 2 (Z-Wave Binary Switch Endpoint) added as $componentLabel"
+	} catch (e) {
+		log.warn "Failed to add endpoint 2 ($desc) as Z-Wave Binary Switch Endpoint - $e"
+	}
+	// Preferences template
+	state.currentPreferencesState = [:]
+	parameterMap.each {
+		state.currentPreferencesState."$it.key" = [:]
+		state.currentPreferencesState."$it.key".value = getPreferenceValue(it)
+		def preferenceName = it.key + "Boolean"
+		settings."$preferenceName" = true
+		state.currentPreferencesState."$it.key".status = "synced"
+	}
+	firstCommand()
+}
+
+def firstCommand(){
+	def commands = []
+	commands << zwave.configurationV1.configurationGet(parameterNumber: 0x01).format()

```

Comment:
Since You are using encap() in other places in this DTH, You might also use it here in this method.

Reply:
I change it like "on()",
use the same encap to send command, it's OK?

Reply:
You may also try to simplify the usage of encap() and cover both: multichannel encapsulation and format() inside encap() method to use it in every call.

Reply:
I try simplify the usage of encap(), and remove secEncap()/command(),
I test this is OK now.